### ISSUE TITLE
Does the output show which primer pair was removed from each read?

### ISSUE THREAD

**Issue by peterjc** [DMU-lilab/pTrimmer#21]
I currently use cutadapt for primer trimming and demultiplexing at the same time. I ask it to produce separate output for each primer pair.

I cannot see that pTrimmer labels the output trimmed reads according to which primer was removed.

Consider the example provided https://github.com/DMU-lilab/pTrimmer/tree/master/Example

```
$ make
...
$ cd Example/
$ ../pTrimmer-1.3.4 -t pair -a data_amplicon.txt  -f data_R1.fq.gz -r data_R2.fq.gz -d out_R1.fastq -e out_R2.fastq
[*] Processing the [thread: 1] ...
[*] Processing the [thread: 2] ...
Total time consume: 0.0(s)

----------------- Summary ------------------------
Total reads processed: 500
Reads have bad primer: 39
Reads have bad quality: 0
Reads successfully trimed and have good quality: 92.20 %
```

(Note there is another "trimed" typo here, see also #18)

The output starts:

```
$ head -n 4 out_R1.fastq 
@ST-E00142:333:H32V7ALXX:2:1101:12784:1731 1:N:0:NTCGAATC
AGTAGGTGAAGAGCTCGTACAGGACGACCCCGAAGCTCCAGACGTCTGACTGGCGAGAGAAGATGTTGTCCGAGAGGGATTCGGGGGCATACCTGGAGAGG
+
JJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
```

```
$ head -n 4 out_R2.fastq 
@ST-E00142:333:H32V7ALXX:2:1101:12784:1731 2:N:0:NTCGAATC
CCTCTCCAGGTATGCCCCCGAATCCCTCTCGGACAACATCTTCTCTCGCCAGTCAGACGTCTGGAGCTTCGGGGTCGTCCTGTACGAGCTCTTCACCTACT
+
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJFFJJJJJJJJJJ
```

The reads have been trimmed, here are the originals:

```
$ cat data_R1.fq.gz | gunzip | grep -A 3 "ST-E00142:333:H32V7ALXX:2:1101:12784:1731"
@ST-E00142:333:H32V7ALXX:2:1101:12784:1731 1:N:0:NTCGAATC
NGCTGCAGCTTTTGTCGCAGTAGGTGAAGAGCTCGTACAGGACGACCCCGAAGCTCCAGACGTCTGACTGGCGAGAGAAGATGTTGTCCGAGAGGGATTCGGGGGCATACCTGGAGAGGGGACAAGGTCTTTGAGATGCGAAGATCGGAA
+
#AAFFJJJJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJFJ
```

and:

```
$ cat data_R2.fq.gz | gunzip | grep -A 3 "ST-E00142:333:H32V7ALXX:2:1101:12784:1731"
@ST-E00142:333:H32V7ALXX:2:1101:12784:1731 2:N:0:NTCGAATC
NCGCATCTCAAAGACCTTGTCCCCTCTCCAGGTATGCCCCCGAATCCCTCTCGGACAACATCTTCTCTCGCCAGTCAGACGTCTGGAGCTTCGGGGTCGTCCTGTACGAGCTCTTCACCTACTGCGACAAAAGCTGCAGCCAGATCGGAA
+
#AAFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJFFJJJJJJJJJJAJ)<FJJJJJJJJJJFFFJ<FJJ7AJF
```

I want to know which primers were removed, the table does not help - note most of the primers report AmpCount zero:

```
$ grep -v "\t0\t" Summary.ampcount | head
FwdPrimer	RevPrimer	AmpCount	AuxInfo
CTTGCTGGATTGAAATTCACTTACACC	GTCCACCATCCTGATAAGGTTAAGG	2	SMAD4,chr18,48604743,48604770,48604856,48604881
TCCTTTAGCTGAAGAATATCTGTCAAAGC	CTCCGAATATCAGCTGCCCTAA	2	ATM,chr11,108139147,108139176,108139250,108139272
ATGTGGTTTCTTGCCTTTGTAAAGTTC	CTGTCTCAGGAGTAGGAAGGATTTTG	1	ATM,chr11,108224398,108224425,108224511,108224537
AGAATCGAAGCGCTACCTGATTC	CTGCCCAACGCACCGAATA	2	CDKN2A,chr9,21974582,21974605,21974699,21974718
TTCTCTTGCTTACATGAACTCTATGTCG	GCCATGATAGGCTCCTGAAAACTAA	4	ATM,chr11,108195999,108196027,108196113,108196138
TGATGCACTTCCATTGACAAGACT	ACAGGTAGAAATAGCCCATGTCATTTAAAA	2	ATM,chr11,108165686,108165710,108165793,108165823
TATCTTTTTACCACAGTTGCACAATATCCT	GCTTTACAGTGAATTGCTGCAACA	2	PTEN,chr10,89692754,89692784,89692869,89692893
CCACGTTTTCGCAGATACATGT	GCAGAACCTCCTCAACACAAGTG	2	JAK3,chr19,17946800,17946822,17946902,17946925
GCGAACTTTGCTAGCTATGTCTAGC	GGAATTTCCCAACAATACAGAGTCTTTG	2	APC,chr5,112154722,112154747,112154820,112154848
```

Turning the question round, which are the two read pairs for this primer pair?:

```
CTTGCTGGATTGAAATTCACTTACACC	GTCCACCATCCTGATAAGGTTAAGG	2	SMAD4,chr18,48604743,48604770,48604856,48604881
```

You might produce FASTQ files for each marker (assuming the name in column 4 can be used in a filename), or append this to the FASTQ description?

**Comment by XLZH** [comment]
Thank you very much! This is a very good idea. I would consider marking the primer information to the header of each read. 
Since I am preparing my PhD thesis, I can only make these modifications after a few months. Thanks again for your suggestion!

**Comment by peterjc** [comment]
Thank you for confirming pTrimmer does not (yet) support this, and good luck with your PhD thesis.

**Comment by peterjc** [comment]
Was closing this as completed an error, or premature?

I see some new commits on the dev branch, but only spelling fixes and work on gzip output. Thanks!

**Comment by XLZH** [comment]
Sorry, it was closed prematurely. A portion of the code is yet to be uploaded, and the feature you mentioned is also included. It should be completed within these days. Thank you for your attention!

**Comment by XLZH** [comment]
@peterjc Hello, The new feature you mentioned, which involves adding primer information to each trimmed READ, has been implemented in pTrimmer-v1.4.0. Please refer to the Usage and Option sections for details. Thank you.

best,
xiaolong zhang

**Comment by peterjc** [comment]
Beware that some FASTQ parsers will reject that kind of output - they expect the "+" line to be blank or match the "@" line.
